### PR TITLE
Add release to compat workflows.

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
The compatibility workflows only target combinations of stable/latest but it is useful to test release candidate against release candidate (PennyLane).

**Description of the Change:**
Add release/release compatibility workflow.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by AmintorDusko — APPROVED** [review]
Nice job!

File: .github/workflows/tests_gpu_cu11.yml [hunk 1]
```diff
@@ -183,14 +183,19 @@ jobs:
           fetch-tags: true
           path: main
 
+      - name: Switch to release build of Lightning
+        if: inputs.lightning-version == 'release'
+        run: |
+          cd main
+          git fetch --all
+          git checkout $(git branch -a --list origin/v*rc* | tail -1)
```
Comment (mlxd) [inline .github/workflows/tests_gpu_cu11.yml:191]: These work with bash, but not with zsh. We can fix that by putting quotes here, and guaranteeing it works on all shells (e.g. if we add MacOS tests here, we need zsh support).

```suggestion
          git checkout $(git branch -a --list "origin/v*rc*" | tail -1)
```

File: .github/workflows/tests_gpu_cu11.yml [hunk 2]
```diff
@@ -233,6 +238,21 @@ jobs:
...
+        with:
+          path: pennylane
+          repository: PennyLaneAI/pennylane
+
+      - name: Switch to release build of PennyLane
+        if: inputs.pennylane-version == 'release'
+        run: |
+          cd pennylane
+          git fetch --all
+          git checkout $(git branch -a --list origin/v*rc* | tail -1)
```
Comment (mlxd) [inline .github/workflows/tests_gpu_cu11.yml:253]:
```suggestion
          git checkout $(git branch -a --list "origin/v*rc*" | tail -1)
```

File: .github/workflows/tests_gpu_kokkos.yml [hunk 3]
```diff
@@ -236,6 +236,13 @@ jobs:
           path: main
           fetch-tags: true
 
+      - name: Switch to release build of Lightning
+        if: inputs.lightning-version == 'release'
+        run: |
+          cd main
+          git fetch --all
+          git checkout $(git branch -a --list origin/v*rc* | tail -1)
```
Comment (mlxd) [inline .github/workflows/tests_gpu_kokkos.yml:244]:
```suggestion
          git checkout $(git branch -a --list "origin/v*rc*" | tail -1)
```

File: .github/workflows/tests_gpu_kokkos.yml [hunk 4]
```diff
@@ -259,6 +266,21 @@ jobs:
...
+        with:
+          path: pennylane
+          repository: PennyLaneAI/pennylane
+
+      - name: Switch to release build of PennyLane
+        if: inputs.pennylane-version == 'release'
+        run: |
+          cd pennylane
+          git fetch --all
+          git checkout $(git branch -a --list origin/v*rc* | tail -1)
```
Comment (mlxd) [inline .github/workflows/tests_gpu_kokkos.yml:281]:
```suggestion
          git checkout $(git branch -a --list "origin/v*rc*" | tail -1)
```

File: .github/workflows/tests_linux.yml [hunk 5]
```diff
@@ -97,6 +97,13 @@ jobs:
           fetch-tags: true
           path: main
 
+      - name: Switch to release build of Lightning
+        if: inputs.lightning-version == 'release'
+        run: |
+          cd main
+          git fetch --all
+          git checkout $(git branch -a --list origin/v*rc* | tail -1)
```
Comment (mlxd) [inline .github/workflows/tests_linux.yml:105]:
```suggestion
          git checkout $(git branch -a --list "origin/v*rc*" | tail -1)
```

File: .github/workflows/tests_linux.yml [hunk 6]
```diff
@@ -120,6 +127,21 @@ jobs:
...
+        with:
+          path: pennylane
+          repository: PennyLaneAI/pennylane
+
+      - name: Switch to release build of PennyLane
+        if: inputs.pennylane-version == 'release'
+        run: |
+          cd pennylane
+          git fetch --all
+          git checkout $(git branch -a --list origin/v*rc* | tail -1)
```
Comment (mlxd) [inline .github/workflows/tests_linux.yml:142]:
```suggestion
          git checkout $(git branch -a --list "origin/v*rc*" | tail -1)
```

File: .github/workflows/tests_linux.yml [hunk 7]
```diff
@@ -224,6 +246,13 @@ jobs:
           fetch-tags: true
           path: main
 
+      - name: Switch to release build of Lightning
+        if: inputs.lightning-version == 'release'
+        run: |
+          cd main
+          git fetch --all
+          git checkout $(git branch -a --list origin/v*rc* | tail -1)
```
Comment (mlxd) [inline .github/workflows/tests_linux.yml:254]:
```suggestion
          git checkout $(git branch -a --list "origin/v*rc*" | tail -1)
```

File: .github/workflows/tests_linux.yml [hunk 8]
```diff
@@ -380,6 +424,13 @@ jobs:
...
```
Comment (mlxd) [inline .github/workflows/tests_linux.yml:432]:
```suggestion
          git checkout $(git branch -a --list "origin/v*rc*" | tail -1)
```

File: .github/workflows/tests_linux.yml [hunk 9]
```diff
@@ -247,6 +276,21 @@ jobs:
...
+        with:
+          path: pennylane
+          repository: PennyLaneAI/pennylane
+
+      - name: Switch to release build of PennyLane
+        if: inputs.pennylane-version == 'release'
+        run: |
+          cd pennylane
+          git fetch --all
+          git checkout $(git branch -a --list origin/v*rc* | tail -1)
```
Comment (mlxd) [inline .github/workflows/tests_linux.yml:291]:
```suggestion
          git checkout $(git branch -a --list "origin/v*rc*" | tail -1)
```

File: .github/workflows/tests_linux.yml [hunk 10]
```diff
@@ -418,6 +469,21 @@ jobs:
...
```
Comment (mlxd) [inline .github/workflows/tests_linux.yml:484]:
```suggestion
          git checkout $(git branch -a --list "origin/v*rc*" | tail -1)
```

File: .github/workflows/tests_without_binary.yml [hunk 11]
```diff
@@ -40,6 +40,13 @@ jobs:
           fetch-tags: true
           path: main
 
+      - name: Switch to release build of Lightning
+        if: inputs.lightning-version == 'release'
+        run: |
+          cd main
+          git fetch --all
+          git checkout $(git branch -a --list origin/v*rc* | tail -1)
```
Comment (mlxd) [inline .github/workflows/tests_without_binary.yml:48]:
```suggestion
          git checkout $(git branch -a --list "origin/v*rc*" | tail -1)
```

File: .github/workflows/tests_without_binary.yml [hunk 12]
```diff
@@ -56,9 +63,25 @@ jobs:
...
+        with:
+          path: pennylane
+          repository: PennyLaneAI/pennylane
+
+      - name: Switch to release build of PennyLane
+        if: inputs.pennylane-version == 'release'
+        run: |
+          cd pennylane
+          git fetch --all
+          git checkout $(git branch -a --list origin/v*rc* | tail -1)
```
Comment (mlxd) [inline .github/workflows/tests_without_binary.yml:82]:
```suggestion
          git checkout $(git branch -a --list "origin/v*rc*" | tail -1)
```

**Review by mlxd — CHANGES_REQUESTED** [review]
Just a simple suggestion to allow support for all shells, local or remote.

**Review by mlxd — APPROVED** [review]
Thanks @vincentmr

**Pull request merged**